pkg/controllers/gateway/gateway_controller.go

```diff
@@ -356,7 +364,7 @@ func (r *GatewayReconciler) getTLSSecrets(ctx context.Context, upstreamGateway *
 			}
 		}
 	}
-	return tlsSecrets, nil
+	return tlsSecrets, listenerTLSErr
 }
```

Comment:
This is probably fine, but it is worth pointing out that this changes the behaviour a bit since nothing is going to get placed on the downstream clusters until all tls secrets that are expected exist in the control plane.

I suppose since we are syncing secrets for gateways it's up to us to decide what the behaviour of our controller should be when secrets don't exist, but would it ever be possible that users just don't want to add their TLS secrets to the control plane cluster like this? Maybe they already have a mechanism to create/sync secrets to their clusters.

Reply:
That is possible I guess. But at this point not something we could say we support. I am trying to think what would trigger a user to want to do it this way? 
If we feel the behaviour here is reasonable, I think it is fine to say we don't support that. If we get enough asks we can change our decision on this ofc

Reply:
Someone wanting to use MGC, but not have it manage TLS. I already have clusters, i already use certmanager in some way to generate TLS secrets for gateways but i want to use MGC/OCM to manage gateways across these clusters.

If/when we do the status aggregation from the downstream clusters you could still achieve this if it is something we wanted to allow.